shell/platform/windows/platform_handler.cc

```diff
@@ -227,6 +227,7 @@ PlatformHandler::PlatformHandler(
       return std::make_unique<ScopedClipboard>();
     };
   }
+  FlutterDesktopPluginRegistrarRegisterTopLevelWindowProcDelegate(engine_->GetRegistrar(), PlatformHandler::WindowProcCallback, this);
```

Comment:
Plugins should be able to override the Windows embedder's behavior. In other words, plugins' top-level window handlers should run before the Windows embedder's top-level window handlers. Please add a test to cover this behavior.

Reply:
That makes sense. Though do we have a way of sending a window message to a top level window for our tests? I'm not aware of anywhere we already do something like this.

Reply:
It doesn't change your valid point, and I realize that's just an example, but saving data on shutdown is an anti-pattern: there are plenty of ways for an app to close without sending a WM_CLOSE that can be handled, so we don't really want to encourage people to do that (it's the reason we eliminated the `exiting` state from the app state design doc).

I think a better example would be so that a plugin could abort the close if it wanted to.